src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -66,17 +77,41 @@ protected ExternalWorkspace run() throws Exception {
             }
 
             String pathOnDisk = computePathOnDisk(physicalPathOnDisk);
-            ExternalWorkspace exws = new ExternalWorkspace(diskPoolId, diskId, pathOnDisk);
-
-            listener.getLogger().println(format("Selected Disk ID '%s' from the Disk Pool ID '%s'", exws.getDiskId(), exws.getDiskPoolId()));
-            listener.getLogger().println(format("The path on Disk is: %s", exws.getPathOnDisk()));
-
-            return exws;
+            exws = new ExternalWorkspace(diskPoolId, diskId, pathOnDisk);
         } else {
             // this is the downstream job
-            // TODO implement
-            return null;
+
+            Item upstreamJob = findUpstreamJob(upstreamName);
+            if (upstreamJob == null) {
+                throw new AbortException(format("There isn't any upstream job associated with '%s'", upstreamName));
+            }
+            Run lastStableBuild = ((Job) upstreamJob).getLastStableBuild();
+            if (lastStableBuild == null) {
+                throw new AbortException(format("'%s' doesn't have any stable build", upstreamName));
+            }
+            if (!(lastStableBuild instanceof WorkflowRun)) {
+                throw new AbortException(format("Build '%s' is not a Pipeline job. Can't read the run actions", lastStableBuild));
+            }
+
+            FlowExecution flowExecution = ((WorkflowRun) lastStableBuild).getExecution();
+            if (flowExecution == null) {
+                throw new Exception("Upstream flow execution is null");
```

Comment:
Mainly I'm using `AbortException` when the build should fail because the user didn't configured the plugin correctly (e.g. missing some parameters)
In this case I've used `Exception` because I don't see what actions the user can take if `flowExecution == null`.

The doc for `getExecution()` says:
>/**
     * Gets the associated execution state.
     * @return non-null after the flow has started, even after finished (but may be null temporarily when about to start, or if starting failed)
     */

I see this as a corner case scenario, very hard (or almost impossible) to reproduce.